libnymea-core/hardware/network/networkdevicediscoveryimpl.cpp

```diff
@@ -608,6 +608,13 @@ void NetworkDeviceDiscoveryImpl::evaluateMonitor(NetworkDeviceMonitorImpl *monit
         return;
     }
 
+    if (!requiresRefresh && monitor->reachable() && currentDateTime <= monitor->lastConnectionAttempt().addSecs(m_rediscoveryInterval)) {
+        // Note: in some cases only a failed ping is able to mark a device as offline if some ARP cache is responding to our ARP requests.
```

Comment:
As far as I understand the problem, this happens because of an ARP Proxy, not because of the ARP cache (The cache is only locally and invalidated by the kernel if there's no traffic after a while). If an ARP request is sent out and actual answer is received even though the device is off, then it's likely an ARP Proxy (likely on the router to optimize wifi performance) that's sending us the answer. Yes, the proxy would take that info from his own ARP cache, so the comment is not technically wrong, but in order to make it more clear what this does, I'd suggest to explicitly saying ARP proxy instead of ARP cache here.

Reply:
check